shell/platform/linux/fl_accessible_text_field.cc

```diff
@@ -0,0 +1,398 @@
+  fl_value_set_string_take(value, "base", fl_value_new_int(base));
+  fl_value_set_string_take(value, "extent", fl_value_new_int(extent));
+
+  g_autoptr(FlStandardMessageCodec) codec = fl_standard_message_codec_new();
+  g_autoptr(GBytes) message =
+      fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec), value, nullptr);
+
+  fl_accessible_node_perform_action(
+      FL_ACCESSIBLE_NODE(self), kFlutterSemanticsActionSetSelection, message);
+}
+
+// Implements GObject::dispose.
+static void fl_accessible_text_field_dispose(GObject* object) {
+  FlAccessibleTextField* self = FL_ACCESSIBLE_TEXT_FIELD(object);
+
+  g_clear_object(&self->buffer);
+
+  G_OBJECT_CLASS(fl_accessible_text_field_parent_class)->dispose(object);
+}
+
+// Implements FlAccessibleNode::set_value.
+static void fl_accessible_text_field_set_value(FlAccessibleNode* node,
+                                               const gchar* value) {
+  g_return_if_fail(FL_IS_ACCESSIBLE_TEXT_FIELD(node));
+  FlAccessibleTextField* self = FL_ACCESSIBLE_TEXT_FIELD(node);
+
+  if (g_strcmp0(gtk_entry_buffer_get_text(self->buffer), value) == 0) {
+    return;
+  }
+
+  gtk_entry_buffer_set_text(self->buffer, value, -1);
+}
+
+// Implements FlAccessibleNode::set_text_selection.
+static void fl_accessible_text_field_set_text_selection(FlAccessibleNode* node,
+                                                        gint base,
+                                                        gint extent) {
+  g_return_if_fail(FL_IS_ACCESSIBLE_TEXT_FIELD(node));
+  FlAccessibleTextField* self = FL_ACCESSIBLE_TEXT_FIELD(node);
+
+  gboolean caret_moved = extent != self->selection_extent;
+  gboolean has_selection = base != extent;
+  gboolean had_selection = self->selection_base != self->selection_extent;
+  gboolean selection_changed = (has_selection || had_selection) &&
+                               (caret_moved || base != self->selection_base);
+
+  self->selection_base = base;
+  self->selection_extent = extent;
+
+  if (selection_changed) {
+    g_signal_emit_by_name(self, "text-selection-changed", nullptr);
+  }
+
+  if (caret_moved) {
+    g_signal_emit_by_name(self, "text-caret-moved", extent, nullptr);
+  }
+}
+
+// Overrides FlAccessibleNode::perform_action.
+void fl_accessible_text_field_perform_action(FlAccessibleNode* self,
+                                             FlutterSemanticsAction action,
+                                             GBytes* data) {
+  switch (action) {
+    case kFlutterSemanticsActionMoveCursorForwardByCharacter:
+    case kFlutterSemanticsActionMoveCursorBackwardByCharacter:
+    case kFlutterSemanticsActionMoveCursorForwardByWord:
+    case kFlutterSemanticsActionMoveCursorBackwardByWord: {
+      g_autoptr(FlValue) extend_selection = fl_value_new_bool(false);
+      g_autoptr(FlStandardMessageCodec) codec = fl_standard_message_codec_new();
+      data = fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec),
```

Comment:
This is going to leak - you need a g_autoptr(GBytes) for this value that overrides the data passed in.